### PR TITLE
Refactor UCM & EuroSAT to VisionClassificationDatasets

### PR DESCRIPTION
**Pull request by isaaccorley**
- Refactored the `UCMerced` and `EuroSAT` datasets to inherit from `VisionClassificationDataset` and removed boilerplate code
- Update tests to increase code cov

Note that I removed some attributes other than the `class_counts`, however they are still available since they are computed under the hood by `torchvision.datasets.ImageFolder`. Stealing from the docs:

```
 Attributes:
    classes (list): List of the class names sorted alphabetically.
    class_to_idx (dict): Dict with items (class_name, class_index).
    imgs (list): List of (image path, class_index) tuples
```

### CODE REVIEW

**Comment by isaaccorley** [comment]
It seems as though numpy received type annotations in 1.21.1, however the last version of numpy supported for Python 3.6 is 1.19 which doesn't have these annotations. Looking for a workaround.

File: torchgeo/datasets/utils.py [hunk 1]
```diff
@@ -375,3 +378,19 @@ def collate_dict(samples: List[Dict[str, Any]]) -> Dict[str, Any]:
                 sample[key] for sample in samples
             ]  # type: ignore[assignment]
     return collated
+
+
+def rasterio_loader(path: str) -> np.ndarray:  # type: ignore[type-arg]
```
Comment (adamjstewart) [inline torchgeo/datasets/utils.py:383]: Could we import `torchvision`'s `default_loader` and rename it `pil_loader` here? Then we can import it and use it elsewhere without wondering what `default_loader` is.

**Review by adamjstewart — DISMISSED** [review]
This looks good to me. I'll let @calebrob6 review to make sure this looks good to him.

**Comment by isaaccorley** [comment]
Updated and rebased.

**Comment by calebrob6** [comment]
This is fine with me!

**Pull request merged**